src/App/Repository/HXL/HXLMetadataRepository.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+/**
+ * Ushahidi HXLMetadata Repository, using Kohana::$config
+ *
+ * @author     Ushahidi Team <team@ushahidi.com>
+ * @package    Ushahidi\Application
+ * @copyright  2014 Ushahidi
+ * @license    https://www.gnu.org/licenses/agpl-3.0.html GNU Affero General Public License Version 3 (AGPL3)
+ */
+namespace Ushahidi\App\Repository\HXL;
+
+use Ushahidi\Core\SearchData;
+use Ushahidi\Core\Entity\HXL\HXLMetadata;
+use Ushahidi\Core\Entity\HXL\HXLMetadataRepository as HXLMetadataRepositoryContract;
+use Ushahidi\App\Repository\OhanzeeRepository;
+
+class HXLMetadataRepository extends OhanzeeRepository implements
+    HXLMetadataRepositoryContract
+{
+    // OhanzeeRepository
+    protected function getTable()
+    {
+        return 'hxl_meta_data';
+    }
+
+    public function getSearchFields()
```

Comment:
Minor ... but if this is write only, then we don't need the search methods, and possibly should be extending the SearchRepository interface in the contract

Reply:
The endpoint is write only. Which doesn't mean we don't search data in that reopo. We use the metadata for sending to hdx and search for dataset_title  to verify if we need to update or create a dataset.